src/components/panel/ManifestLabel.tsx

```diff
@@ -0,0 +1,82 @@
+import { FC, useEffect, useState } from 'react'
+import { Popover, PopoverContent, PopoverTrigger } from '@/components/ui/popover.tsx'
+import { Button } from '@/components/ui/button.tsx'
+import { useTranslation } from 'react-i18next'
+import { CircleAlert } from 'lucide-react'
+
+import { useDataStore } from '@/store/DataStore.tsx'
+import { usePanel } from '@/contexts/PanelContext.tsx'
+import { apiRequest } from '@/utils/api.ts'
+
+
+interface ManifestLabelProps {
+  selectedManifest: Manifest | null,
+  updateSelectedManifest: (newManifest: Manifest | null) => void
+  setShowItemModal: (show: boolean) => void,
+}
+
+const ManifestLabel: FC<ManifestLabelProps> = ({ selectedManifest, updateSelectedManifest, setShowItemModal }) => {
```

Comment:
setShowItemModal shouldn't be part of that components logic, label comps should only open/close and select, any logic beyond should be handled in the parent comp